contracts/gateway-internal.yaml

```diff
@@ -0,0 +1,184 @@
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: >
+            **Causes:**
+              * Downstream system(s) is down.
+              * Unhandled exceptions.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  paths:
+  /bridges/101/services:
+    post:
+      tags:
+        - registration
+      summary: creation of bridge services entries in DB by admin
+      description: >
+        This API is meant for creating bridge services entries in database.
+      parameters:
+        - $ref: "#/components/parameters/authorization"
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/BridgeServiceRequest'
+          application/xml:
+            schema:
+              $ref: '#/components/schemas/BridgeServiceRequest'
+      responses:
+        '204':
+          description: Bridge Services entries added successfully
+        '400':
+          description: >
+            Invalid request, required attributes not provided
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: >
+            **Causes:**
+              * Unauthorized request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: >
+            **Causes:**
+              * Downstream system(s) is down.
+              * Unhandled exceptions.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+components:
+  parameters:
+    authorization:
+      name: Authorization
+      in: header
+      required: true
+      description: Access token which was issued after successful login with gateway auth server.
+      schema:
+        type: string
+  schemas:
+    BridgeRegistrationRequest:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        url:
+          type: string
+        active:
+          type: boolean
+        blocklisted:
+          type: boolean
+      xml:
+        name: BridgeRegistrationRequest
+    BridgeServiceRequest:
+      type: object
+      properties:
+        bridgeId:
+          type: string
+        services:
+          type: array
+          items:
+            $ref: '#/components/schemas/BridgeService'
```

Comment:
Since you are posting against /internal/bridge/101/services, ideally you do not need a nested attribute "services", you can just post an array of "service" to it.

Reply:
But I still need service id, type & active status to do an upsert into db right?